classes/gateways/class.pmprogateway_stripe.php

```diff
@@ -4300,55 +4300,50 @@ public function refund( &$order, $transaction_id = null ) {
 			}
 		}
 
-		//get the charge
+		$success = false;
+
+		//attempt refund
 		try {
-			$charge = Stripe_Charge::retrieve( $transaction_id );
-		} catch ( \Throwable $e ) {
-			$charge = false;
-		} catch ( \Exception $e ) {
-			$charge = false;
-		}
+			
+			$secretkey = pmpro_getOption( 'stripe_secretkey' );
 
-		//can't find the charge?
-		if ( empty( $charge ) ) {
-			$order->status     = "error";
-			$order->errorcode  = "";
-			$order->error      = "";
-			$order->shorterror = "";
+			// If they are not using legacy keys, get Stripe Connect keys for the relevant environment.
+			if ( ! self::using_legacy_keys() && empty( $secretkey ) ) {
+				if ( pmpro_getOption( 'gateway_environment' ) === 'live' ) {
+					$secretkey = pmpro_getOption( 'live_stripe_connect_secretkey' );
+				} else {
+					$secretkey = pmpro_getOption( 'sandbox_stripe_connect_secretkey' );
+				}
+			} 
 
-			return false;
-		}
+			$client = new Stripe_Client( $secretkey );
+			$refund = $client->refunds->create( [
+				'charge' => $transaction_id,
+			] );
 
-		//attempt refund
-		try {
-			$refund = $charge->refund();
-		} catch ( \Throwable $e ) {
-			$order->errorcode  = true;
-			$order->error      = __( "Error: ", 'paid-memberships-pro' ) . $e->getMessage();
-			$order->shorterror = $order->error;
+			$success = true;
+			
+			global $current_user;
 
-			return false;
-		} catch ( \Exception $e ) {
-			$order->errorcode  = true;
-			$order->error      = __( "Error: ", 'paid-memberships-pro' ) . $e->getMessage();
-			$order->shorterror = $order->error;
+			$notes = $order->notes;
+			$order->notes = '\n\n'.sprintf( __('Order successfully refunded on %1$s for transaction  ID %2$s by %3$s', 'paid-memberships-pro' ), date_i18n('Y-m-d H:i:s'), $transaction_id, $current_user->display_name );	
 
-			return false;
-		}
+			if ( $refund->status == 'succeeded' ) {
+				$order->status = 'refunded';					
+			} else {
+				$order->notes = trim( $order->notes ) .' '. __('An error occured while attempting to process this refund.', 'paid-memberships-pro' );
```

Comment:
The `trim()` should wrap the whole new note and not just the previous note.

```suggestion
				$order->notes = trim( $order->notes .' '. __( 'An error occured while attempting to process this refund.', 'paid-memberships-pro' ) );
```

Or

```suggestion
				$order->notes = $order->notes .' '. __( 'An error occured while attempting to process this refund.', 'paid-memberships-pro' );
				$order->notes = trim( $order->notes );
```